app/controllers/user/sessions_controller.rb

```diff
@@ -0,0 +1,47 @@
+class User::SessionsController < Devise::SessionsController
+  def create
+    if session.has_key?(:user_sign_in_uid)
+      self.resource = User.find(session[:user_sign_in_uid])
+      session.delete(:user_sign_in_uid)
```

Comment:
Since `Hash#delete` returns the object that's deleted, this can be simplified as:

```suggestion
      self.resource = User.find(session.delete(:user_sign_in_uid))
```